re/service.go

```diff
@@ -34,10 +43,100 @@ const (
 	Monthly
 )
 
+func (rt ReccuringType) String() string {
+	switch rt {
+	case Daily:
+		return "daily"
+	case Weekly:
+		return "weekly"
+	case Monthly:
+		return "monthly"
+	default:
+		return "none"
+	}
+}
+
+func (rt ReccuringType) MarshalJSON() ([]byte, error) {
+	return json.Marshal(rt.String())
+}
+
+func (rt *ReccuringType) UnmarshalJSON(data []byte) error {
+	var s string
+	if err := json.Unmarshal(data, &s); err != nil {
+		return err
+	}
+
+	switch s {
+	case "daily":
+		*rt = Daily
+	case "weekly":
+		*rt = Weekly
+	case "monthly":
+		*rt = Monthly
+	case "none":
+		*rt = None
+	default:
+		return ErrInvalidRecurringType
+	}
+	return nil
+}
+
 type Schedule struct {
-	Time            []time.Time   `json:"date,omitempty"`
-	RecurringType   ReccuringType `json:"recurring_type"`
-	RecurringPeriod uint          `json:"recurring_period"` // 1 meaning every Recurring value, 2 meaning every other, and so on.
+	StartDateTime   time.Time     `json:"start_datetime"`           // When the schedule becomes active
+	RecurringTime   []time.Time   `json:"recurring_time,omitempty"` // Specific times for the rule to run
+	RecurringType   ReccuringType `json:"recurring_type"`           // None, Daily, Weekly, Monthly
+	RecurringPeriod uint          `json:"recurring_period"`         // 1 meaning every Recurring value, 2 meaning every other, and so on.
+}
+
+func (s Schedule) MarshalJSON() ([]byte, error) {
```

Comment:
In the DB, we store it as TIMESTAMP (not TIMESTAMPZ) because we don't want to take DBMS server timezone into account; we'll rather always work with UTC internally. That being said, we need to take care all the inputs are taken **with** timezone and converted to UTC.